authorization/role_assignment_service_blackbox_test.go

```diff
@@ -0,0 +1,240 @@
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), resourceRefUnrelated)
+
+	roleRef, err := testsupport.CreateTestRole(s.Ctx, s.DB, *areaResourceType, "collab")
+	require.NoError(s.T(), err)
+
+	var createdIdentityRoles []role.IdentityRole
+
+	// creating an AssignedRole for a different resource - not expected
+	// to show up in search results.
+	identityRoleRefUnrelated, err := testsupport.CreateTestIdentityRole(s.Ctx, s.DB, *resourceRefUnrelated, *roleRef)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), identityRoleRefUnrelated)
+
+	// creating an AssignedRole for a specific resource
+	// which will be queried for listing all assigned roles
+	identityRoleRef1, err := testsupport.CreateTestIdentityRole(s.Ctx, s.DB, *resourceRef, *roleRef)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), identityRoleRef1)
+	createdIdentityRoles = append(createdIdentityRoles, *identityRoleRef1)
+
+	// creating an AssignedRole for a specific resource
+	// which will be queried for listing all assigned roles
+	identityRoleRef2, err := testsupport.CreateTestIdentityRole(s.Ctx, s.DB, *resourceRef, *roleRef)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), identityRoleRef1)
+	createdIdentityRoles = append(createdIdentityRoles, *identityRoleRef2)
+
+	identityRoles, err := s.assignmentService.ListByResource(s.Ctx, identityRoleRef1.Resource.ResourceID)
+	require.NoError(t, err)
+	require.Len(t, identityRoles, 2)
+	require.Equal(t, true, s.checkExists(*identityRoleRef1, identityRoles, false))
+	require.Equal(t, true, s.checkExists(*identityRoleRef2, identityRoles, false))
+
+}
+
+func (s *roleAssignmentServiceBlackboxTest) TestGetIdentityRoleByResourceNotFound() {
+	t := s.T()
+	identityRole, err := testsupport.CreateRandomIdentityRole(s.Ctx, s.DB)
+	require.NoError(t, err)
+	require.NotNil(t, identityRole)
+
+	identityRoles, err := s.assignmentService.ListByResource(s.Ctx, uuid.NewV4().String())
+	require.Error(t, errors.NotFoundError{})
+	require.Equal(t, 0, len(identityRoles))
+}
+
+func (s *roleAssignmentServiceBlackboxTest) checkExists(createdRole role.IdentityRole, pool []role.IdentityRole, isInherited bool) bool {
+	for _, retrievedRole := range pool {
+		if retrievedRole.IdentityRoleID.String() == createdRole.IdentityRoleID.String() {
+			s.compare(createdRole, retrievedRole, isInherited)
+			return true
+		}
+	}
+	return false
+}
+
+func (s *roleAssignmentServiceBlackboxTest) compare(createdRole role.IdentityRole, retrievedRole role.IdentityRole, isInherited bool) bool {
+	require.Equal(s.T(), createdRole.IdentityRoleID.String(), retrievedRole.IdentityRoleID.String())
+	require.Equal(s.T(), createdRole.IdentityID.String(), retrievedRole.Identity.ID.String())
+	require.Equal(s.T(), createdRole.Role.Name, retrievedRole.Role.Name)
+
+	if isInherited {
+		require.NotNil(s.T(), createdRole.Resource.ParentResourceID)
+		require.Equal(s.T(), *createdRole.Resource.ParentResourceID, *createdRole.Resource.ParentResourceID)
```

Comment:
This looks like it is comparing exactly the same value, which will always be true?

Reply:
fixed b5a87de